data/users/externalldap.py

```diff
@@ -16,7 +17,9 @@
 _DEFAULT_PAGE_SIZE = 1000
 # setting config LDAP_FOLLOW_REFERRALS: 0 to disable referral lookups
 _DEFAULT_REFERRALS = True
-
+_DEFAULT_KEEPALIVE_IDLE = 10
+_DEFAULT_KEEPALIVE_INTERVAL = 5
+_DEFAULT_KEEPALIVE_PROBES = 3
```

Comment:
Just curious as to how we figured out these were sane defaults. 
What are the values when not explicitly set?

Reply:
the default values are 

KEEPALIVE_IDLE = 7200 # 2 hours
KEEPALIVE_INTERVAL = 75 # seconds
KEEPALIVE_PROBES = 9 # until connection get's closed.

with the `quay-operator` killing the POD within ~2 minutes if the health checks fail, it seems obvious to set them low as considered being internal network connectivity the size of 64b is worth mentioning and recovery should be possible.
The settings reflect a connection after 10 seconds of being IDLE will send very 5 seconds a keepalive and drop/close the connection after worst case 15 seconds no response received. If meanwhile data passes the connection, keepalive will reset for the 10 seconds IDLE.

Reply:
do you want me to change/add config.yaml parameters for it ?

Reply:
I don't think that's necessary as you're currently just updating the default values. If you want to make it configurable, you would have to add these to https://github.com/quay/quay/blob/master/data/users/__init__.py#L56-L90.

Reply:
well I guess we can save that for a different PR  :dancing_women: 
thanks for your input and support, will update the PR accordingly